services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/RocksDbKeyValueStorage.java

```diff
@@ -61,12 +64,26 @@ private RocksDbKeyValueStorage(
     RocksDbUtil.loadNativeLibrary();
     try {
       stats = new Statistics();
+
+
       options =
           new Options()
               .setCreateIfMissing(true)
               .setMaxOpenFiles(rocksDbConfiguration.getMaxOpenFiles())
-              .setTableFormatConfig(rocksDbConfiguration.getBlockBasedTableConfig())
+              .setTableFormatConfig(new BlockBasedTableConfig()
+                  .setBlockSize(16 * 1024)
+                  .setPinL0FilterAndIndexBlocksInCache(true)
+                  .setCacheIndexAndFilterBlocks(true) // make this an --X option
+                  .setFilter(new BloomFilter(10, true))
+                  .setFormatVersion(2)
+                  .setBlockCache(new LRUCache(rocksDbConfiguration.getCacheCapacity()))
+              )
+              .optimizeForPointLookup(rocksDbConfiguration.getCacheCapacity() >> 20) // API wants MBs
```

Comment:
FYI ` * org.rocksdb.util.SizeUnit.MB` would be more readable than `>> 20` here.

Reply:
done